src/app/public/modules/datepicker/datepicker-input.directive.ts

```diff
@@ -333,18 +333,23 @@ export class SkyDatepickerInputDirective
 
     const value = event.target.value;
 
-    let errors: ValidationErrors;
-
+    let errors = Object.assign({}, this.control.errors);
     if (this.isDateStringValid(value)) {
+      // Reset the invalid state.
+      if (errors.skyDate) {
+        delete errors.skyDate.invalid;
+      }
+    } else {
+      errors.skyDate = {
+        invalid: true
+      };
+    }
+
+    // Set errors to null if the object is empty.
+    if (Object.keys(errors).length === 0) {
```

Comment:
This is good, but you should also delete the `errors.skyDate` if it's empty. See:
![captured](https://user-images.githubusercontent.com/12497062/69351047-a98d2680-0c48-11ea-885d-a268c5a4b11d.gif)